cli/src/main/java/org/infinispan/cli/commands/rest/Schema.java

```diff
@@ -41,16 +36,85 @@ public boolean isHelp() {
    }
 
    @Override
-   protected CompletionStage<RestResponse> exec(ContextAwareCommandInvocation invocation, RestClient client, org.infinispan.cli.resources.Resource resource) {
-      if ((upload != null) && (args.size() != 1)) {
-         throw Messages.MSG.illegalCommandArguments();
-      } else if ((upload == null) && (args.size() != 2)) {
-         throw Messages.MSG.illegalCommandArguments();
-      }
-      if (upload == null) {
-         return client.schemas().put(args.get(0), args.get(1));
-      } else {
-         return client.schemas().put(args.get(0), RestEntity.create(MediaType.TEXT_PLAIN, new File(upload.getAbsolutePath())));
+   public CommandResult exec(ContextAwareCommandInvocation invocation) {
+      // This command serves only to wrap the sub-commands
+      invocation.println(invocation.getHelpInfo());
+      return CommandResult.FAILURE;
+   }
+
+   @CommandDefinition(name = "upload", description = "Uploads schemas to the server")
+   public static class Upload extends RestCliCommand {
+      @Argument(required = true, description = "The name of the schema")
+      String name;
+
+      @Option(completer = FileOptionCompleter.class, shortName = 'f', description = "The protobuf file to upload", required = true)
+      Resource file;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      protected CompletionStage<RestResponse> exec(ContextAwareCommandInvocation invocation, RestClient client, org.infinispan.cli.resources.Resource resource) {
+         return client.schemas().put(name, RestEntity.create(MediaType.TEXT_PLAIN, new File(file.getAbsolutePath())));
+      }
+   }
+
+   @CommandDefinition(name = "ls", description = "Lists available schemas")
+   public static class Ls extends RestCliCommand {
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      protected CompletionStage<RestResponse> exec(ContextAwareCommandInvocation invocation, RestClient client, org.infinispan.cli.resources.Resource resource) {
+         return client.schemas().names();
+      }
+   }
+
+   @CommandDefinition(name = "remove", aliases = "rm", description = "Removes a schema")
```

Comment:
for some reason, the aliases never worked
```
[pedro-desktop-25956@cluster//containers/default/schemas]> schema rm person.proto
schema 'rm' is not a schema command. See 'schema --help'.
```

Reply:
Yeah, it's an aesh bug. It will be fixed with my picocli version